drivers/w1/w1_max32.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2023-2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control/adi_max32_clock_control.h>
+
+#include <wrap_max32_owm.h>
+
+#define DT_DRV_COMPAT adi_max32_w1
+
+LOG_MODULE_REGISTER(w1_max32, CONFIG_W1_LOG_LEVEL);
+
+struct max32_w1_config {
+	struct w1_master_config w1_config;
+	mxc_owm_regs_t *regs;
+	const struct pinctrl_dev_config *pctrl;
+	const struct device *clock;
+	struct max32_perclk perclk;
+	int internal_pullup;
+	int external_pullup;
+	int long_line_mode;
```

Comment:
using 12 bytes to represent configuration options with 4 significant bits seems not extra optimized, but it's const any benefit would depend on implementation so I won't block, just leave a note.

Reply:
Converted as uint8_t, thanks.